src/Suggestor/Suggestor.ts

```diff
@@ -183,7 +191,7 @@ function addDatesSuggestions(
         // a max number. We want the max number to be around half the total allowed matches, to also allow
         // some global generic matches (e.g. task components) to find their way to the menu
         const minMatch = 1;
-        const maxGenericSuggestions = 5;
+        const maxGenericSuggestions = settings.autoSuggestMaxItems - 1;
```

Comment:
I wanted this variable to be derived from the settings max suggestions, so the tests could run without being limited to an arbitrary 5. I changed this to max items - 1 because the default max suggestions is 6 and it was set to 5, but now that I'm seeing it in the context of the PR, I notice the comment above says it should be half of the matches! Let me update this 😅

Reply:
I updated it to be half of the maximum suggestions, but then I thought a bit more about it... 

Once you've added a priority, a created date and a due date, there are only 3 general suggestions left (start, scheduled, recurrence). I you then want to add, say, a recurrence, first you select it (only 2 general suggestions left), and then it only suggests max / 2 recurrence suggestions, which is 3 with the default setting of 6 suggestions. That means that we're only showing 3 + 2 = 5 items total, when the user wanted 6 suggestions. That means we're not being as helpful as we could be! It gets more obvious if the user has selected to see more suggestions than the default.

With a little refactor, that is, moving the general suggestions generation first so we know how many there are (and adding them later), we can determine how much space we actually have for the date/recurrence suggestions. I have it working, but it's a bit outside the scope of the original issue though. Would you like me to add this to the PR? This is what it looks like: https://github.com/ldanet/obsidian-tasks/commit/e45603ec4cbf2136f01651e8a0c1777204a971f0

Reply:
Thank you for this suggestion. Sorry but I'm not going to have any cycles to look at this until Saturday at the earliest...

Reply:
No rush at all!